client/api/dbs/dbs.py

```diff
@@ -0,0 +1,198 @@
+from collections import UserList
+from datetime import datetime
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.security import generate_password_hash, check_password_hash
+
+db = SQLAlchemy()
+
+
+class Users(db.Model):
+    """User table to handle the user logins"""
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(100), unique=True)
+    password = db.Column(db.String(100))
+    is_active = db.Column(db.Boolean)
+    is_authenticated = db.Column(db.Boolean)
+
+    def get_id(self):
+        """get user id"""
+        return self.id
+
+    def signin(uname, pwd):
+        """Adding a new user to the User database"""
+        user = Users.query.filter_by(username=uname).first()
+        if not user:
+            return 1
+        elif not check_password_hash(user.password, pwd):
+            return 2
+        else:
+            return user
+
+    def signup(uname, pwd):
+        """Adding a new user to the User database"""
+        new_user = Users(
+            username=uname,
+            password=generate_password_hash(pwd, method="sha256"),
+            is_active=True,
+            is_authenticated=True,
+        )
+        db.session.add(new_user)
+        db.session.commit()
+
+    def get_username(user_id):
+        """Returns the username for a given user_id"""
+        user = Users.query.filter_by(id=user_id).first()
+        if not user:
+            return "annonymous"
+        else:
+            return user.username
+
+
+class UserProfile(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("Users.id"), nullable=False)
+    email = db.Column(db.String(100), nullable=False)
+    first_name = db.Column(db.String(50))
+    last_name = db.Column(db.String(50))
+    profile_pic = db.Column(db.BLOB)
+
+    def updateUserProfile(user_id, email, first_name, last_name, profile_pic):
+        profile = UserProfile.query.filter_by(user_id=user_id).first()
+        if not profile:
+            return "Profile Not Found"
+        else:
+            if email != "":
+                profile.email = email
+            if first_name != "":
+                profile.first_name = first_name
+            if last_name != "":
+                profile.last_name = last_name
+            if profile_pic is not None:
+                profile.profile_pic = profile_pic
+            db.session.commit()
+            return "Profile Successfully Updated"
+
+    def getUserProfile(user_id):
+        userProfile = UserProfile.query.filter_by(user_id=user_id).first()
+        if not userProfile:
+            return "Profile Not Found"
+        else:
+            return userProfile
+
+
+class MediaTypes(db.Model):
```

Comment:
What is the media types database for? Is this for like storing results from the search and if thats the case do we have any need to store it?

Reply:
Resolved in person :)